### PR TITLE
feat: re-write setting up environment, add using vscode section

### PR DESCRIPTION
**Pull request by kitsonk**

### CODE REVIEW

File: getting_started/setup_your_environment.md [hunk 1]
```diff
@@ -324,6 +193,162 @@ endif
...
+roots = [".git"]
+command = "deno"
+args = ["lsp"]
+
+[language.deno.initialization_options]
+enable = true
+lint = true
+```
+
+### Environment variables
```
Comment (bartlomieju) [inline getting_started/setup_your_environment.md:234]: Missing section?

Comment (bartlomieju) [reply getting_started/setup_your_environment.md:234]: Actually not missing, but it's listed below, please remove

**Review by bartlomieju — APPROVED** [review]
LGTM, very useful information. Please address one comment below before merging.

File: getting_started/setup_your_environment.md [hunk 2]
```diff
@@ -106,88 +51,59 @@ more information on how to set-up your JetBrains IDE for Deno, read
...
+are a user of JetBrains and Deno, voicing your support can continue help
+JetBrains prioritize support.
+
+#### vim/Neovim
+
+Deno is well supported on both [vim](https://www.vim.org/) and
+[Neovim](https://neovim.io/) via
+[coc.nvim](https://github.com/neoclide/coc.nvim) and
+[ALE](https://github.com/dense-analysis/ale). coc.nvim offers plugins to
+integrate to the Deno language server while ALE supports it _out of the box_.
```
Comment (David-Else) [inline getting_started/setup_your_environment.md:66]: The official config for Neovim 0.5 is provided by https://github.com/neovim/nvim-lspconfig and is detailed at https://github.com/neovim/nvim-lspconfig/blob/master/CONFIG.md#denols

Comment (TeoDev1611) [reply getting_started/setup_your_environment.md:66]: I added this in #57

**Pull request merged**